service/controller/app/v1/resource/chart/create_test.go

```diff
@@ -0,0 +1,198 @@
+				},
+			},
+			expectedChart: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "kubernetes-prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Config: v1alpha1.ChartSpecConfig{
+						ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
+							Name:            "giant-swarm-config",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+						Secret: v1alpha1.ChartSpecConfigSecret{
+							Name:            "giant-swarm-secret",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+					},
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+		},
+		{
+			name: "case 1: chart already existed",
+			currentResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "kubernetes-prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Config: v1alpha1.ChartSpecConfig{
+						ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
+							Name:            "giant-swarm-config",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+						Secret: v1alpha1.ChartSpecConfigSecret{
+							Name:            "giant-swarm-secret",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+					},
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "kubernetes-prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Config: v1alpha1.ChartSpecConfig{
+						ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
+							Name:            "giant-swarm-config",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+						Secret: v1alpha1.ChartSpecConfigSecret{
+							Name:            "giant-swarm-secret",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+					},
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			var err error
+			var kc *kubeconfig.KubeConfig
```

Comment:
We should add a `kubeconfigtest` package like we have for `helmclient`.

https://github.com/giantswarm/helmclient/blob/c93791a1e9e60a08f2ee48394d416a726c5e00f2/helmclienttest/helmclient.go#L31

That way we only need `kubeconfigtest.New()` and it keeps the tests shorter.

Reply:
I added the new file https://github.com/giantswarm/app-operator/blob/e5cd4788e943a1b24b03705972b439fcc2213d6b/kubeconfigtest/kubeconfig.go

Reply:
@tomahawk28 Thanks for adding the `kubeconfigtest` package but sorry my comment could have been clearer.

This should be created in the kubeconfig repo and then we just import it. That way we can use it wherever we use the library. https://github.com/giantswarm/kubeconfig

It's a cleanup task so if you want to do that in a separate PR that's fine with me.

Reply:
Oops, I got it, You're right we should do that on next PR.